Allow easy customization of EmbeddedMongo DownloadConfig

mgmeiner opened the pull request:
This PR introduces 2 new configuration properties to allow easy customization of the DownloadPath of a mongodb. 

- `spring.mongodb.embedded.download.path` = custom download url
- `spring.mongodb.embedded.download.user-agent` = custom User-Agent HTTP-Header for the download request

We need this due to some special security rules in our company which prevents us from downloading the binaries directly from mongodb. With this PR it is possible to easy override the default download-path and point it to our own server. 
Also it is now possible to easy create a completely custom DownloadConfig by simply providing a bean which implements `IDownloadConfig`

pivotal-issuemaster commented:
@mgmeiner Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-boot&pullRequestId=15496)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-boot&pullRequestId=15496) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

pivotal-issuemaster commented:
@mgmeiner Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-boot&pullRequestId=15496)!

mgmeiner commented:
@wilkinsona I like your idea on this and you're right: introducing new properties for the download config which is indeed a very specific case - is not justified. I updated the PR and implemented it how you suggested. I named the Customizer `EmbeddedMongoDownloadConfigBuilderCustomizer` to match the naming of the other classes in the package.

mgmeiner commented:
not sure why the tests are failing.. Locally all tests have passed without error

wilkinsona commented:
That's a [known issue](https://github.com/flapdoodle-oss/de.flapdoodle.embed.mongo/issues/246) that may be caused by an underlying bug in the JDK. I'd like Embedded Mongo to retry in that case, but the suggestion didn't get any traction.

snicoll commented:
@mgmeiner thank you for making your first contribution to Spring Boot. This is now merged in `master` with a [polish commit](https://github.com/spring-projects/spring-boot/commit/6ba1f40e595cd18deca470791f354494ac546ada).
